components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/PredictStreamProcessor.java

```diff
@@ -203,6 +274,12 @@ private void populateFeatureAttributeMapping() throws Exception {
         }
     }
 
+    private void logError(int modelId, Exception e) {
+        log.error("Error while retrieving ML-model : " + modelStorageLocations[modelId], e);
+        throw new ExecutionPlanCreationException(
```

Comment:
This method called for 4 different exceptions but the message should be the same, except the exception.
It's used to avoid code duplication.